app/Config/Database.php

```diff
@@ -185,6 +185,7 @@ class Database extends Config
             'datetime' => 'Y-m-d H:i:s',
             'time'     => 'H:i:s',
         ],
+        'enableNestedTransactions' => true,
```

Comment:
I'm not sure this config key is good for this change. I would like to hear others' opinions. > @codeigniter4/core-team @codeigniter4/database-team 

Because we have already the `transException()` method to change CI4's transaction behavior.
https://codeigniter4.github.io/CodeIgniter4/database/transactions.html#throwing-exceptions

Reply:
Such a change is indeed problematic. I don't know how to reconcile exception handling with savepoints.

Anyway, maybe we should use something like `enableSavepointTransactions`? Also, can we refer users to the same section in the user guide describing the differences between nested transactions and save point transactions?

Last thing. I haven't checked from which versions, but SQLSRV and OCI8 also support savepoints - possibly with different syntax, but that doesn't change the fact that support should be provided.

Reply:
Yes that one came to mind too. In the same vein I personally don't like the current behaviour of BaseConnection::query, which rerolls ALL (outer) transactions when a single nested query fails, when DBDebug + transException is set: https://github.com/codeigniter4/CodeIgniter4/blob/develop/system/Database/BaseConnection.php#L658-L670, as they might still be caugt and rerolled.

If the point is to make sure we don't leave transactions in limbo, we should do this on a cleanup step (__destruct() or something).

Reply:
Addionally, I actually like having DBDebug on, so more exceptions are thrown generally, so database errors not just get ignored, or the database even gets in a weird unexpected state.

Reply:
@michalsn What am I missing regarding the differences between nested transactions and savepoints, https://sqlite.org/lang_savepoint.html, look for "There are several ways of thinking about the RELEASE command", the first one sounds like a nested transaction, the second one is the way i like to think about it. Also they seem to be equivalent in some way?

Regarding SQLsrv + OCI8, I don't have access to those systems (except here in CI, when you guys run it), so this will be very labourious and spammy in here.

Reply:
We have `docker-compose.yaml` for all supported databases:
https://github.com/codeigniter4/devkit/tree/develop/src/docker